### PR TITLE
Style guide for struct variables in book

### LINKED ISSUE #248: Style guide for struct variables in book
### Motivation

Add section for

```rust
struct Structure {
    amount: u64
}

fn action(amount: u64) {
    let structure = Structure { amount };
}
```

so that the following is avoided

```rust
struct Structure {
    amount: u64
}

fn action(value: u64) {
    let amount = value;
    let structure = Structure { amount: value };
    let structure = Structure { amount: amount };
}

### PR DESCRIPTION
**Pull request by Braqzen**
## Type of change

- Documentation

## Changes

The following changes have been made:

- Added section for initializing struct fields with the same variable name

## Related Issues

Closes #248

### CODE REVIEW

**Comment by Braqzen** [comment]
I believe the following [PR](https://github.com/FuelLabs/sway-libs/pull/27) has somehow caused a leak which is breaking CI. I don't understand how this has happened since there has been no new release nor has the version changed in this repo.

**Review by matt-user — DISMISSED** [review]

**Comment by bitzoic** [comment]
> I believe the following [PR](https://github.com/FuelLabs/sway-libs/pull/27) has somehow caused a leak which is breaking CI. I don't understand how this has happened since there has been no new release nor has the version changed in this repo.

This is confusing me as well. The use of releases was intended to prevent this exact situation.

File: .docs/contributing-book/src/documentation/project-quality/documentation/code/style.md [hunk 1]
```diff
@@ -46,6 +46,28 @@ The general approach is to omit a type if the compiler does not throw an error h
 {{#include ../../../../code/style_guide/src/lib.sw:type_annotation}}
 ```
 
+## Field initialization shorthand
```
Comment (bitzoic) [inline .docs/contributing-book/src/documentation/project-quality/documentation/code/style.md:49]:
```suggestion
## Field Initialization Shorthand
```
I believe the other headings in the file are also capitalized

**Review by bitzoic — CHANGES_REQUESTED** [review]
Anchors 💪

File: .docs/contributing-book/src/documentation/project-quality/documentation/code/style.md (hunk 1 above)
Comment (Braqzen) [reply .docs/contributing-book/src/documentation/project-quality/documentation/code/style.md:49]: I sometimes wonder which part to capitalize for emphasis or if everything should be done.

Comment (matt-user) [reply .docs/contributing-book/src/documentation/project-quality/documentation/code/style.md]: What I generally follow is [title capitalization](https://www.grammarly.com/blog/capitalization-in-the-titles/#:~:text=According%20to%20most%20style%20guides,part%20of%20speech%20they%20are.) rules for markdown headers.

**Review by matt-user — APPROVED** [review]

**Pull request merged**